src/features/mcda/calculations/operations.ts

```diff
@@ -0,0 +1,30 @@
+import type maplibregl from 'maplibre-gl';
+
+export interface IsomorphMath<T> {
+  add: (x: T, y: T) => T;
+  sub: (x: T, y: T) => T;
+  mult: (x: T, y: T) => T;
+  div: (x: T, y: T) => T;
+  log: (x: T) => T;
+  sqrt: (x: T) => T;
+}
+
+export type MapExpression = maplibregl.Expression;
+
+export class MapMath implements IsomorphMath<MapExpression> {
+  add = (x: MapExpression, y: MapExpression) => ['+', x, y] as MapExpression;
+  sub = (x: MapExpression, y: MapExpression) => ['-', x, y] as MapExpression;
+  mult = (x: MapExpression, y: MapExpression) => ['*', x, y] as MapExpression;
+  div = (x: MapExpression, y: MapExpression) => ['/', x, y] as MapExpression;
+  log = (x: MapExpression) => ['ln', x] as MapExpression;
```

Comment:
log is usually base 10 (or custom base), ln is base e. Which one here?

Reply:
interesting, but in js [Math.log](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log) it is a logarithm with base e